src/peft/utils/hotswap.py

```diff
@@ -87,14 +311,47 @@ def hotswap_adapter_from_state_dict(model, state_dict, adapter_name, parameter_p
 
     # actual swapping
     for key, new_val in state_dict.items():
+        module_name = ".".join(key.split(".")[:-3])
+        module = model.get_submodule(module_name)
+
+        # swap alpha/scaling
+        r_key = get_pattern_key(config.rank_pattern.keys(), key)
+        alpha_key = get_pattern_key(config.alpha_pattern.keys(), key)
+        rank = config.rank_pattern.get(r_key, config.r)
+        alpha = config.alpha_pattern.get(alpha_key, config.lora_alpha)
+        if config.use_rslora:
+            scaling = alpha / math.sqrt(rank)
+        else:
+            scaling = alpha / rank
+        _update_scaling(module, adapter_name=adapter_name, scaling=scaling)
+
+        # swap actual weights
         # no need to account for potential _orig_mod in key here, as torch handles that
         old_val = attrgetter(key)(model)
-        if is_compiled:
-            # Compiled models don't work with swap_tensors because there are weakrefs for the tensor. It is unclear if
-            # this workaround could not cause trouble but the tests indicate that it works.
+        if not is_compiled:
+            torch.utils.swap_tensors(old_val, new_val)
+            continue
+
+        # Compiled models don't work with swap_tensors because there are weakrefs for the tensor. It is unclear if
+        # this workaround could not cause trouble but the tests indicate that it works.
+        if old_val.shape == new_val.shape:
             old_val.data = new_val.data
         else:
-            torch.utils.swap_tensors(old_val, new_val)
+            if old_val.dim() != 2:
+                # TODO conv2d
```

Comment:
```suggestion
                # TODO conv1d & conv2d
```

I don't think that there's a difference in implementation but to be sure...

Reply:
Since I'd say we leave out conv1d for now, we don't need a TODO comment for it. I'll probably complete the conv2d part soon if we can confirm with diffusers that it works.